lib/Bindings/Python/ESIModule.cpp

```diff
@@ -0,0 +1,54 @@
+//===- ESIModule.cpp - ESI API pybind module ------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "DialectModules.h"
+
+#include "circt-c/Dialect/ESI.h"
+#include "circt/Dialect/ESI/ESIDialect.h"
```

Comment:
It seems to work based on the test case, but I think this pybind11 module is diverging from the notion of only layering on top of the CAPI. Specifically, the new `resolvePortNames`, and the existing `buildESIWrapper` functions are not part of the ESI CAPI.

I'm not sure if this is just lucky right now in terms of how the linking and RTTI and everything line up, or what. It might be worth adding a wrapper in the ESI CAPI layer to go through, instead of pointing to the dialect header/library directly.

Reply:
Yeah, I'm not familiar with the linking issues. I'm going through the C API for all the MLIR types, where there's a strong chance that linking issues could come up.  The calls you're talking about are relatively simple C++ calls that are linked in the same way they would be in a C API. The C API function would actually be C++ code with C name mangling (and perhaps different linking mechanisms) so I'm not sure this is necessary.

@stellaraccident Can you comment on this? Will we run into linking issues by calling C++ code directly from a pybind11 wrapped function wherein I use the C API for Mlir* types, but don't go through a C API for simple C++ function calls? Near as I can tell, npcomp does that [here ](https://github.com/llvm/mlir-npcomp/blob/2ab62aec12fd0ea0fbe75df2bfb52307e160a478/lib/Backend/RefJIT/PythonModule.cpp#L74)(and presumably elsewhere).